File: ConsultoresGISGeoServiceBase/XML/XMLScheduleGetter.vb
Imports System.Xml

Namespace XML
    ''' <summary>
    ''' Obtiene el itinerario del archivo Config.xml
    ''' </summary>
    ''' <remarks></remarks>
    Public Class XMLScheduleGetter
        Inherits XMLGetter

        ''' <summary>
        ''' Obtiene una lista de horarios para los cuales está configurado el servicio
        ''' </summary>
        ''' <returns>Lista de horarios en formato HH:mm</returns>
        ''' <remarks></remarks>
        Public Shared Function getSchedule() As List(Of String)
            Dim schedule As New List(Of String)

            Try
                For Each nodo As XmlElement In getSingleXMLElement("schedule").GetElementsByTagName("time")
                    schedule.Add(nodo.GetAttribute("start"))
                Next
            Catch ex As Exception
                Logger.Logger.Error(ex)
            End Try

            Return schedule
        End Function
    End Class
End Namespace

File: GeoServicesSharp/XML/XMLScheduleGetter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace GeoServicesSharp.XML
{
    /// <summary>
    /// Obtiene el itinerario del archivo Config.xml
    /// </summary>
    class XMLScheduleGetter : XMLGetter
    {
        /// <summary>
        /// Obtiene una lista de horarios para los cuales está configurado el servicio
        /// </summary>
        /// <returns>Lista de horarios en formato HH:mm</returns>
        public static List<string> getSchedule()
        {
            List<string> schedule = new List<string>();

            try
            {
                foreach (XmlElement nodo in getSingleXMLElement("schedule").GetElementsByTagName("time"))
                    schedule.Add(nodo.GetAttribute("start"));
            }
            catch (Exception ex)
            {
                Logger.Logger.Error(ex);
            }

            return schedule;
        }
    }
}